packages/ic-response-verification/src/verification/verify_request_response_pair.rs

```diff
@@ -272,14 +273,23 @@ fn v2_verification(
     );
 
     match are_hashes_valid {
-        true => Ok(VerificationInfo {
-            response: Some(VerifiedResponse {
-                status_code: Some(response.status_code),
-                headers: response_headers.headers,
-                body: response.body,
-            }),
-            verification_version: 2,
-        }),
+        true => {
+            let certificate_headers: Vec<(String, String)> = vec![(
+                CERTIFICATE_HEADER_NAME.to_string(),
+                response_headers.certificate.unwrap(),
```

Comment:
Maybe use `expect(..)` instead of `unwrap()`?
Or even better, return an error if `certificate` is `None`?